custom/icds_reports/utils/aggregation_helpers/distributed/deprecation.py

```diff
@@ -0,0 +1,161 @@
+from dateutil.relativedelta import relativedelta
+from django.db import connections
+
+from corehq.apps.userreports.util import get_table_name
+from corehq.sql_db.connections import get_icds_ucr_citus_db_alias
+from custom.icds_reports.const import DASHBOARD_DOMAIN
+from custom.icds_reports.utils.aggregation_helpers import get_prev_table, transform_day_to_month
+
+
+class TempPrevTables(object):
+
+    DROP_QUERY = """
+    DROP TABLE IF EXISTS "{prev_table}";
+    DROP TABLE IF EXISTS "{prev_local}";
+    """
+
+    def drop_temp_tables(self, alias):
+        data = {
+            'prev_table': get_prev_table(alias),
+            'prev_local': f"{alias}_prev_local",
+        }
+        with connections[get_icds_ucr_citus_db_alias()].cursor() as cursor:
+            cursor.execute(self.DROP_QUERY.format(**data))
+
+    def create_temp_tables(self, alias, table, day):
+        data = {
+            'prev_table': get_prev_table(alias),
+            'prev_local': f"{alias}_prev_local",
+            'prev_month': day,
+            'current_table': table,
+            'alias': alias
+        }
+        with connections[get_icds_ucr_citus_db_alias()].cursor() as cursor:
+            cursor.execute(self.CREATE_QUERY.format(**data))
+
+    def make_all_tables(self, day):
+        raise NotImplementedError
+
+
+class TempPrevUCRTables(TempPrevTables):
+
+    CREATE_QUERY = """
+    CREATE UNLOGGED TABLE "{prev_table}" (LIKE "{current_table}");
+    SELECT create_distributed_table('{prev_table}', 'supervisor_id');
+    INSERT INTO "{prev_table}" (SELECT * FROM "{current_table}");
+    CREATE INDEX "idx_rationalization_date_{alias}" ON "{prev_table}" USING hash (location_rationalisation_date);
```

Comment:
nit: should choose between british and american spelling if we're sticking with this terminology

Reply:
These will all be reassignment soon, but yep. They were all american until I realized the spec called for the case property to use the british spelling. Fixed all those cases but must have missed this